File: api/svg/svg/Services/SVGService/SVGService.cs
using svg.Data.FileSystem;
using svg.Models;

namespace svg.Services.SVGService
{
  public class SVGService : ISVGService
  {

    ISVGRepository sVGRepository;
    public SVGService(ISVGRepository _sVGRepository)
    {
      sVGRepository = _sVGRepository;
    }
    public SVG GetSVG()
    {
      return sVGRepository.Get();
    }

    public SVG UpdateSVG(SVG svg)
    {
      return sVGRepository.Update(svg);
    }
  }
}

File: src/app/services/svg-service/svg.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class SvgService {

  baseURL = 'https://localhost:3000';

  constructor(private http: HttpClient) { }

  getSVG(): Observable<any> {
    const url = `${this.baseURL}/api/svg`;
    return this.http.get(url);
  }

  updateSVG(width: number, height: number): Observable<any> {
    const url = `${this.baseURL}/api/svg`;
    const data = { width, height };
    return this.http.post(url, data);
  }
}
